ksqldb-engine/src/main/java/io/confluent/ksql/services/KafkaConsumerGroupClientImpl.java

```diff
@@ -117,10 +115,11 @@ public void deleteConsumerGroups(final Set<String> groups) {
       ExecutorUtil.executeWithRetries(
           () -> adminClient.get().deleteConsumerGroups(groups).all().get(),
           e -> (e instanceof RetriableException)
-              || (e instanceof GroupNotEmptyException && retryCount.getAndIncrement() < 5),
-          () -> Duration.of(3, ChronoUnit.SECONDS)
+              || (e instanceof GroupNotEmptyException),
+          (retry) -> Duration.of(3L * retry, ChronoUnit.SECONDS),
+          10
```

Comment:
Second change: allow more retries. It was previously hard-coded in ExecutorUtil to 5, now it will be 10. So we would previously only wait up to 12 seconds before throwing an exception, and now we will wait for 135 seconds.